torch/onnx/__init__.py

```diff
@@ -139,23 +142,50 @@ def _run_symbolic_method(op_name, symbolic_fn, args):
         raise
 
 
-def _add_attribute(node, key, value):
+def _is_onnx_list(value):
+    if not isinstance(value, string_classes) and not torch.is_tensor(value) and isinstance(value, collections.Iterable):
+        return True
+    return False
+
+
+def _add_attribute(node, key, value, aten):
     """ initializes the right attribute based on type of value """
     m = attr_pattern.match(key)
     if m is None:
         raise IndexError((
             "Invalid attribute specifier '{}' names " +
             " must be suffixed with type, e.g. 'dim_i' or 'dims_i'").format(key))
     name, kind = m.group(1), m.group(2)
-    if not isinstance(value, string_classes) and not torch.is_tensor(value) and isinstance(value, collections.Iterable):
+    if _is_onnx_list(value):
         kind += "s"
-    return getattr(node, kind + '_')(name, value)
+    if aten:
+        if torch.is_tensor(value):
+            # Turn tensor attribute to float/int attribute.
```

Comment:
Remember the rule for comments: don't say what, say why. The reason *why* we have to do it this way, is because the Caffe2-ATen integration code assumes that tensor attributes are always single-valued.